src/components/ConfirmationModal/ConfirmationModal.tsx

```diff
@@ -30,6 +31,14 @@ export type Props = PropsWithSpread<
      * Function to perform the action prompted by the modal.
      */
     onConfirm: (event: MouseEvent<HTMLElement>) => void;
+    /**
+     * Is the confirm button loading.
+     */
+    confirmButtonLoading?: boolean;
+    /**
+     * Disable the confirm button.
```

Comment:
I would use the same descriptions that we have for the `ActionButton` here:
```suggestion
     * Whether the confirm button should be in the loading state.
     */
    confirmButtonLoading?: boolean;
    /**
     * Whether the confirm button should be disabled.
```